src/Chainweb/Version.hs

```diff
@@ -508,3 +488,87 @@ randomChainId :: HasChainwebVersion v => v -> IO ChainId
 randomChainId v = (!!) (toList cs) <$> randomRIO (0, length cs - 1)
   where
     cs = chainIds v
+
+-- -------------------------------------------------------------------------- --
+-- Properties of Chainweb Versions
+-- -------------------------------------------------------------------------- --
+
+-- -------------------------------------------------------------------------- --
+-- Graph
+
+chainwebVersionGraph :: ChainwebVersion -> ChainGraph
+chainwebVersionGraph (Test g) = g
+chainwebVersionGraph (TimedConsensus g) = g
+chainwebVersionGraph (PowConsensus g) = g
+chainwebVersionGraph (TimedCPM g) = g
+chainwebVersionGraph (FastTimedCPM g) = g
+chainwebVersionGraph Development = petersonChainGraph
+chainwebVersionGraph Testnet02 = petersonChainGraph
+
+instance HasChainGraph ChainwebVersion where
+    _chainGraph = chainwebVersionGraph
+    {-# INLINE _chainGraph #-}
+
+-- -------------------------------------------------------------------------- --
+-- POW Parameters
+
+-- | The gap in SECONDS that we desire between the Creation Time of subsequent
+-- blocks in some chain.
+--
+newtype BlockRate = BlockRate Seconds
+
+-- | The Proof-of-Work `BlockRate` for each `ChainwebVersion`. This is the
+-- number of seconds we expect to pass while a miner mines on various chains,
+-- eventually succeeding on one.
+--
+blockRate :: ChainwebVersion -> Maybe BlockRate
+blockRate Test{} = Nothing
+blockRate TimedConsensus{} = Just $ BlockRate 4
+blockRate PowConsensus{} = Just $ BlockRate 10
+blockRate TimedCPM{} = Just $ BlockRate 4
+-- 120 blocks per hour, 2,880 per day, 20,160 per week, 1,048,320 per year.
+blockRate FastTimedCPM{} = Just $ BlockRate 1
+blockRate Development = Just $ BlockRate 30
+-- 120 blocks per hour, 2,880 per day, 20,160 per week, 1,048,320 per year.
+blockRate Testnet02 = Just $ BlockRate 30
```

Comment:
Why is it more useful for these functions to live in this module?

Reply:
The idea is to eventually have all specifications of chainweb versions in a single place. That that when a new version is added or an old one removed, there is a well-defined place where all changes have to be made.

Reply:
Eventually, there should be data structure ala

```haskell
data ChainwebVersionSpecification = ChainwebVersionSpecification
    { _versionSpecName :: !T.Text
    , _versionSpecBlockRate :: !BlockRate
   , _versionSpecEpochSize :: !EpochSize
    , _versionSpecGraph :: !ChainwebGraph
    {- etc. -}
    }
```

And then there would be one constant value of this type defined for each version, e.g. in a module `Chainweb.Version.Specifications`.

That would also make it easier to define and run some basic consistency checks in a test suite.

Reply:
Ah I misread - I thought those had been moved to `BlockHeader`.